src/v/cluster/security_frontend.cc

```diff
@@ -323,23 +323,35 @@ security_frontend::get_bootstrap_user_creds_from_env() {
     }
 
     ss::sstring creds_str = creds_str_ptr;
-    auto colon = creds_str.find(":");
-    if (colon == ss::sstring::npos || colon == creds_str.size() - 1) {
+    std::string_view creds = creds_str_ptr;
+    std::vector<std::string_view> parts;
+    parts.reserve(3);
+    boost::algorithm::split(parts, creds, [](char c) { return c == ':'; });
+    if (
+      parts.size() != 2 || parts.size() != 3 || parts[0].empty()
+      || parts[1].empty()) {
         // Malformed value.  Do not log the value, it may be malformed
         // but it is still a secret.
         vlog(
           clusterlog.warn,
-          "Invalid value of {} (expected \"username:password\")",
+          "Invalid value of {} (expected \"username:password:mechanism\")",
           bootstrap_user_env_key);
         return {};
     }
-
-    auto username = security::credential_user{creds_str.substr(0, colon)};
-    auto password = creds_str.substr(colon + 1);
-    auto credentials = security::scram_sha256::make_credentials(
-      password, security::scram_sha256::min_iterations);
+    std::variant<security::scram_sha256, security::scram_sha512> scram;
+    if (
+      parts.size() == 3
+      && parts[2] == security::scram_sha512_authenticator::name) {
```

Comment:
Pretty sure this will require an include.

Reply:
I don't see an additional include in other files that use this, but maybe I'm missing it? For instance: https://github.com/redpanda-data/redpanda/blob/51a22204777c5d28f8be6550701e59459c1a90c8/src/v/kafka/server/server.cc#L516

Reply:
src/v/security/scram_authenticator.h